### PR TITLE
feat: add cargo crd

### PR DESCRIPTION
**Pull request by cd1989**
**What this PR does / why we need it**:

Create new custom resource `cargo`, which will be used to manage cargo auth in clusters.

https://docs.google.com/document/d/1jnh1tCBaLXYTFTM0REOuteWaHZ8Jz35FCBcfbGTtWA4/edit

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/cargo/v1/types.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+type Cargo struct {
+	// Metadata for the resource, like kind and apiversion
+	metav1.TypeMeta `json:",inline"`
+
+	// Metadata for the particular object, including name, namespace, labels, etc
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Custom resource specification
+	Spec CargoSpec `json:"spec"`
+}
```
Comment (li-ang) [inline pkg/apis/cargo/v1/types.go]: 结构体内的空行都去掉吧

File: pkg/apis/cargo/v1/types.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type Cargo struct {
```
Comment (li-ang) [inline pkg/apis/cargo/v1/types.go]: 结构体前面建议加个注释，解释一下这个 CRD 是干什么

File: pkg/apis/cargo/v1/types.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+type CargoSpec struct {
+	Name   string `json:"name"`
+	Domain string `json:"domain"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type CargoList struct {
```
Comment (li-ang) [inline pkg/apis/cargo/v1/types.go]: 建议加 `// CargoList describes an array of Cargo instances.` 这样一条注释。

**Comment by cd1989** [comment]
@li-ang 非常有建设性的建议，我就加一个 devops 吧

File: pkg/apis/cargo/v1/types.go (hunk 3 above)
Comment (cd1989) [reply pkg/apis/cargo/v1/types.go]: ok

File: pkg/apis/cargo/v1/types.go (hunk 2 above)
Comment (cd1989) [reply pkg/apis/cargo/v1/types.go]: ok

File: pkg/apis/cargo/v1/types.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	// Metadata for the resource, like kind and apiversion
+	metav1.TypeMeta `json:",inline"`
+
+	// Metadata for the particular object, including name, namespace, labels, etc
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Custom resource specification
+	Spec CargoSpec `json:"spec"`
+}
+
```
[the remark this answers is not in the document]
Comment (cd1989) [reply pkg/apis/cargo/v1/types.go]: ok

**Pull request merged**